### PR TITLE
[readme] Add README

### PR DESCRIPTION
**Pull request by willjvsmith**
* Makes the README not empty

See what it says on the tin

### CODE REVIEW

**Review by jaylees14 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,14 @@
 # spec-golang
-A sample service written in Go
+
+This repository holds the example microservice backend written for the development of [Temple](https://github.com/TempleEight/temple), and is therefore written with the goal of being as general as possible.
+
+This example models the backend for a Tinder-like application, with services for users and matches, as well as a separate service for authentication.
```
Comment (c-harding) [inline README.md]:
```suggestion
This example models the backend for a dating application, with services for users and matches, as well as a separate service for authentication.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,14 @@
...
+* PostgreSQL is used as the backing datastore
```
Comment (c-harding) [inline README.md:7]: [markdownlint/MD032](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#MD032)
```suggestion

* PostgreSQL is used as the backing datastore
```

**Review by Lewiky — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,2 +1,14 @@
...
+* Kong acts as the API gateway and load balancer
+* Prometheus and Grafana are used for gathering and displaying metrics
+* OpenAPI is used to generate frontend APIs
+* Scripts are supplied for orchestration with either Kubernetes or Docker Compose
+
+For more information on the architecture we settled on, check out the docs:
+* [System Architecture](https://templeeight.github.io/temple-docs/docs/arch/system)
```
Comment (c-harding) [inline README.md:15]: [markdownlint/MD032](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#MD032)
```suggestion

* [System Architecture](https://templeeight.github.io/temple-docs/docs/arch/system)
```

**Review by c-harding — APPROVED** [review]
Looks good, just my linter complaining about a lack of spacing around the lists

**Pull request merged**